uuid.go

```diff
@@ -344,6 +345,29 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+// MarshalJSON marshalls the NullUUID as nil or the nested UUID
+func (u NullUUID) MarshalJSON() ([]byte, error) {
+	if u.Valid == false {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(u.UUID)
+}
+
+// UnmarshalJSON unmarshalls a NullUUID
+func (u *NullUUID) UnmarshalJSON(b []byte) error {
+	if bytes.Equal(b, []byte("null")) {
```

Comment:
Should we mark this as an invalid UUID? It seems like sending `null` could be a very explicit way to do the same thing, without sending 16 NULL bytes.

Reply:
I am not sure what you mean here, we must fill the NullUUID struct. Unless we change it to contain a pointer, we can't set u.UUID to nil, can we?